homeassistant/components/microbees/cover.py

```diff
@@ -0,0 +1,116 @@
+"""Cover integration microBees."""
+from typing import Any
+
+from microBeesPy import Actuator
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+
+from .const import DOMAIN
+from .coordinator import MicroBeesUpdateCoordinator
+from .entity import MicroBeesEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Config entry."""
+    coordinator: MicroBeesUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ].coordinator
+
+    async_add_entities(
+        MBCover(
+            coordinator,
+            bee_id,
+            next(filter(lambda x: x.deviceID == 551, bee.actuators)).id,
+            next(filter(lambda x: x.deviceID == 552, bee.actuators)).id,
+        )
+        for bee_id, bee in coordinator.data.bees.items()
+        if bee.productID == 47
+    )
+
+
+class MBCover(MicroBeesEntity, CoverEntity):
+    """Representation of a microBees cover."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN | CoverEntityFeature.STOP | CoverEntityFeature.CLOSE
+    )
+
+    def __init__(self, coordinator, bee_id, actuator_up_id, actuator_down_id) -> None:
+        """Initialize the microBees cover."""
+        super().__init__(coordinator, bee_id)
+        self.actuator_up_id = actuator_up_id
+        self.actuator_down_id = actuator_down_id
+        self._attr_is_closed = None
```

Comment:
There is no way to determine the current state?

Reply:
no, there's not

Reply:
Ok, but we still need to set `_attr_is_closed` to an assumed state after we have executed the open or close command. Otherwise the state would always be unknown.

Reply:
> Ok, but we still need to set `_attr_is_closed` to an assumed state after we have executed the open or close command. Otherwise the state would always be unknown.

we cannot determine or assume the status because you can also use manual mode (without our device) to open or close and we have no way to have the actual status.

Reply:
You could take a look at the Comelit cover platform, which also cannot retrieve the current status. I'd rather have a status that doesn't match the physical status of the device 100% (due to manual intervention on the device or something) than a constant "unknown" status which is not a good user experience in my eyes.

Reply:
Actually we don't like the Comelit solution. We prefer customer can open and close in every case and stop. For example if I want to open just a little I can do it stopping whenever I want. Assuming the actual status avoid this and can cause inconsistent status in case of manual intervention or something else.
In our opinion this is the best user experience and freedom for the use of the device

Reply:
I talked to another core dev about this, and it's fine to have the permanent "unknown" state. So feel free to mark the PR as ready for review.